interviewTest/Task1.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ This is simple list component with the following expected functionality:
+
+ 1. All items in the list are rendered correctly on the screen
+ 2. Pressing the button sorts the list and re-renders it
+ 3. The justifyContent prop is used to determine the Flex CSS layout
+
+ There are issues with this implementation. Can you see them?
+*/
+import React, { useState } from "react";
+import { Button, StyleSheet, Text, View } from "react-native";
+
+const data = ["Rick", "Morty", "Jerry", "Summer"];
+
+type Props = {
+  justifyContent: "flex-start" | "flex-end" | "center";
+};
+
+export const MyList = (props: Props) => {
+  const { justifyContent } = props;
+
+  const [listItems, setListItems] = useState(data);
+
+  const handleSort = () => setListItems(listItems.sort());
```

Comment:
This implementation will cause the UI to do wasteful work on the second, third etc times that the user presses the the "Sort items" button. The work will be wasteful because the result of `listItems.sort()` is always going to be the same since the sorting criteria never changes, so the app will be trying to sort an already-sorted list. For this component with non-changing sort criteria and a static list, my suggestion is to have a sorted copy of the list outside the component before it is even rendered.

Then, you can have the code look like:
```suggestion
const data = ["Rick", "Morty", "Jerry", "Summer"];
// JS does in-place sorting, so calling `data.sort()` will actually sort `data` itself.
// Plus, I want `sortedData` to be an object of its own, not just a reference to `data`.
const sortedData = [...data].sort();

export const MyList = (props: Props) => {
  ...
  const handleSort = () => setListItems(sortedData);
```